src/pages/FlagCommentPage.tsx

```diff
@@ -0,0 +1,185 @@
+            furtherDetailsIcon: Expensicons.FlagLevelTwo,
+        },
+        {
+            severity: CONST.MODERATION.FLAG_SEVERITY_HARASSMENT,
+            name: translate('moderation.harassment'),
+            icon: Expensicons.FlagLevelThree,
+            description: translate('moderation.harassmentDescription'),
+            furtherDetails: translate('moderation.levelThreeResult'),
+            furtherDetailsIcon: Expensicons.FlagLevelThree,
+        },
+        {
+            severity: CONST.MODERATION.FLAG_SEVERITY_ASSAULT,
+            name: translate('moderation.assault'),
+            icon: Expensicons.FlagLevelThree,
+            description: translate('moderation.assaultDescription'),
+            furtherDetails: translate('moderation.levelThreeResult'),
+            furtherDetailsIcon: Expensicons.FlagLevelThree,
+        },
+    ];
+
+    const getActionToFlag = useCallback(() => {
+        let reportAction = reportActions?.[`${route.params.reportActionID.toString()}`];
+
+        // Handle threads if needed
+        if (reportAction?.reportActionID === undefined) {
+            reportAction = parentReportActions?.[report?.parentReportActionID ?? ''] ?? undefined;
+        }
+
+        return reportAction;
+    }, [report, reportActions, route.params.reportActionID, parentReportActions]);
+
+    const flagComment = (severity: ValueOf<typeof CONST.MODERATION>) => {
+        let reportID: string | undefined = getReportID(route);
+        const reportAction = getActionToFlag();
+        const parentReportAction = parentReportActions?.[report?.parentReportActionID ?? ''] ?? undefined;
+
+        // Handle threads if needed
+        if (ReportUtils.isChatThread(report) && reportAction?.reportActionID === parentReportAction?.reportActionID) {
+            reportID = ReportUtils.getParentReport(report)?.reportID;
+        }
+
+        if (reportAction && ReportUtils.canFlagReportAction(reportAction, reportID)) {
+            Report.flagComment(reportID ?? '', reportAction, severity);
+        }
+
+        Navigation.dismissModal();
+    };
+
+    const severityMenuItems = severities.map((item) => (
+        <MenuItem
+            key={`${item.severity}`}
+            shouldShowRightIcon
+            title={item.name}
+            description={item.description}
+            onPress={Session.checkIfActionIsAllowed(() => flagComment(item.severity))}
+            style={[styles.pt2, styles.pb4, styles.ph5, styles.flexRow]}
+            furtherDetails={item.furtherDetails}
+            furtherDetailsIcon={item.furtherDetailsIcon}
+        />
+    ));
+
+    return (
+        <ScreenWrapper
+            includeSafeAreaPaddingBottom={false}
+            testID={FlagCommentPage.displayName}
+        >
+            {({safeAreaPaddingBottomStyle}) => {
+                const actionToFlag = getActionToFlag();
+                const shouldShowFlagComment = actionToFlag && ReportUtils.shouldShowFlagComment(actionToFlag, report);
+
```

Comment:
You can get rid of that changes making `getActionToFlag` return null instead of undefined